From efd7df6bf878df1ae3b7f91c7dd1fec01737bc85 Mon Sep 17 00:00:00 2001
From: Adlink CCoE <dinesh.v@adlinktech.com>
Date: Mon, 20 Mar 2023 14:27:37 +0530
Subject: [PATCH]  Avoid ACPI probe due to resource conflict with BIOS for
 accessing SMBUS register

---
 drivers/i2c/busses/i2c-i801.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/i2c-i801.c b/drivers/i2c/busses/i2c-i801.c
index 74d343d1a36b..8c81b458e7a9 100644
--- a/drivers/i2c/busses/i2c-i801.c
+++ b/drivers/i2c/busses/i2c-i801.c
@@ -1784,8 +1784,9 @@ static int i801_probe(struct pci_dev *dev, const struct pci_device_id *id)
 		return -ENODEV;
 	}
 
-	if (i801_acpi_probe(priv))
-		return -ENODEV;
+	// Resource conflict with BIOS accessing SMBUS register
+/*	if (i801_acpi_probe(priv))
+		return -ENODEV; */
 
 	err = pcim_iomap_regions(dev, 1 << SMBBAR, DRV_NAME);
 	if (err) {
-- 
2.25.1

