diff --git a/drivers/i2c/busses/i2c-i801.c b/drivers/i2c/busses/i2c-i801.c
index ba8df2fde1b2..faa33337d178 100644
--- a/drivers/i2c/busses/i2c-i801.c
+++ b/drivers/i2c/busses/i2c-i801.c
@@ -1504,6 +1504,7 @@ static int i801_probe(struct pci_dev *dev, const struct pci_device_id *id)
 	priv->adapter.dev.parent = &dev->dev;
 	ACPI_COMPANION_SET(&priv->adapter.dev, ACPI_COMPANION(&dev->dev));
 	priv->adapter.retries = 3;
+	priv->adapter.nr=20;
 	mutex_init(&priv->acpi_lock);
 
 	priv->pci_dev = dev;
@@ -1576,8 +1577,8 @@ static int i801_probe(struct pci_dev *dev, const struct pci_device_id *id)
 		return -ENODEV;
 	}
 
-	if (i801_acpi_probe(priv))
-		return -ENODEV;
+//	if (i801_acpi_probe(priv)) /*Adlink: removing check for resource conflict with BIOS accessing SMBUS register*/
+//		return -ENODEV;
 
 	err = pcim_iomap_regions(dev, 1 << SMBBAR,
 				 dev_driver_string(&dev->dev));
