diff --git a/drivers/net/ethernet/intel/igc/igc_main.c b/drivers/net/ethernet/intel/igc/igc_main.c
index 4cb7294727ba..64a941156642 100644
--- a/drivers/net/ethernet/intel/igc/igc_main.c
+++ b/drivers/net/ethernet/intel/igc/igc_main.c
@@ -5538,8 +5538,10 @@ static int igc_save_qbv_schedule(struct igc_adapter *adapter,
 	u32 start_time = 0, end_time = 0;
 	size_t n;
 
-	if (!qopt->enable)
-		return igc_tsn_clear_schedule(adapter);
+	if (!qopt->enable) {
+		adapter->base_time = 0;
+		return 0;
+	}
 
 	if (adapter->base_time)
 		return -EALREADY;
@@ -6050,7 +6052,7 @@ static int igc_probe(struct pci_dev *pdev,
 
 	igc_ptp_init(adapter);
 
-	igc_tsn_clear_schedule(adapter);
+//	igc_tsn_clear_schedule(adapter);
 
 	/* reset the hardware with the new settings */
 	igc_reset(adapter);
